packages/@lwc/engine-core/src/framework/modules/events.ts

```diff
@@ -53,16 +19,10 @@ function createAllEventListeners(vnode: VNode) {
         return;
     }
 
-    const listener: VNodeEventListener = ((vnode as InteractiveVNode).listener = createListener());
-    listener.vnode = vnode;
-
-    let name;
-    for (name in on) {
-        addEventListener(elm, name, listener);
+    for (const name in on) {
+        const handler = on[name];
+        addEventListener(elm, name, (event: Event) => {
```

Comment:
Some questions:

1. before we used to have 1 handler for all events, now we have 1 handler per event (as many handlers as events). Pls, confirm.
2. why can we skip `on && on[type]` check?
3. why not just do `renderer.addEventListener(elm, name, handler);`?

Reply:
For # 2, if we've gotten to line 23, then `on` must be defined (`for (const name in undefined) {}` would never iterate otherwise).

For # 3 I agree unless there's a good reason.

For # 1 we're still calling `Node.prototype.addEventListener` the same number of times, so I would imagine there isn't a big perf implication. Although we are creating multiple functions when we used to just create one. (# 3 would fix that.)

Reply:
about 2), `const a = {a: 'b', b: undefined };for (const n in a) { console.log(n); };` does loop over prop `b`, but i think they are always defined since `vnode.data.on` contains the template event listeners, which are our bound event listeners.